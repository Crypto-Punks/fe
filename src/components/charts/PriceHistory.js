import { Line } from 'react-chartjs-2';
import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { getPriceHistory } from '../../services/currencies';
import moment from 'moment';
import styles from './PriceHistory.css';

const PriceHistory = ({ id, historyDuration }) => {
  const [intervals, setIntervals] = useState([]);
  const [datasets, setDatasets] = useState([]);

  const priceHistoryOptions = {
    title: {
      display: true,
      text: 'Price History'
    },
    animation: {
      duration: 1000,
      easing: 'linear',
      showlines: true
    }, 
    responsive: true,
    
    
  };

  useEffect(() => {
    getPriceHistory(id, historyDuration)
      .then(priceHistory => {
        setIntervals(priceHistory.map(item => moment(item.time).format('MMM Do YYYY')));
        setDatasets(priceHistory.map(item => item.priceUsd));
      });
  }, [historyDuration]);

  return (
    <div className={styles.PriceHistory}>
      <Line data={{
        labels: intervals,
        datasets: [{ label: 'price', data: datasets, backgroundColor: 'rgba(255, 0, 0, 0.1)' }]
      }} options={priceHistoryOptions} />
    </div>
  );
};

PriceHistory.propTypes = {
  id: PropTypes.string.isRequired,
  historyDuration: PropTypes.string.isRequired
};

export default PriceHistory;
