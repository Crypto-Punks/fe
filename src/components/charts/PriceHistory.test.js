import React from 'react';
import { shallow } from 'enzyme';
import PriceHistory from './PriceHistory';

const data = {
  id: 'test',
  historyDuration: 'test'
};

describe('PriceHistory component', () => {
  it('renders PriceHistory', () => {
    const wrapper = shallow(<PriceHistory { ...data} />);
    expect(wrapper).toMatchSnapshot();
  });
});
