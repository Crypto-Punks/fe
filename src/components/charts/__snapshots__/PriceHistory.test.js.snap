// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PriceHistory component renders PriceHistory 1`] = `
<div
  className="PriceHistory"
>
  <Line
    data={
      Object {
        "datasets": Array [
          Object {
            "backgroundColor": "rgba(255, 0, 0, 0.2)",
            "borderColor": "rgba(0, 0, 0, 0.2)",
            "borderWidth": 0.5,
            "data": Array [],
            "label": "price",
          },
        ],
        "labels": Array [],
      }
    }
    options={
      Object {
        "animation": Object {
          "duration": 1000,
          "easing": "linear",
          "showlines": true,
        },
        "responsive": true,
        "title": Object {
          "display": true,
          "text": "Price History",
        },
      }
    }
  />
</div>
`;
